Yank/pipeline.py

```diff
@@ -1531,7 +1579,32 @@ def _setup_system(self, system_file_path, pack, alchemical_charge,
         tleap.export_script(base_file_path + '.leap.in')
 
         # Run tleap and log warnings
-        warnings = tleap.run()
+        # Handle common errors we know of
+        try:
+            warnings = tleap.run()
+        except RuntimeError as e:
+            # Try to find the log file
+            m = re.search('Check log file (.*)', str(e))
+            if m is not None:
+                log_file_string = m.group(1)
+                try:
+                    with open(log_file_string, 'r') as log_file:
+                        log_file_data = log_file.read()
+                    # Analyze log file for water mismatch
+                    if "EP - OW" in log_file_data:
+                        # Found mismatch water and missing parameters
+                        error = ("Solvent {}: "
+                                 "It looks like the water used has virtual sites, but "
+                                 "missing parameters.\nMake sure your leap parameters "
+                                 "use the correct water model as specified by "
+                                 "solvent_model.\nPlease see the log file for LEaP "
+                                 "for more info as there may be more wrong:\n{}")
+                        raise RuntimeError(error.format(solvent_id, e))
```

Comment:
Should we put this code in `tleap.run()` rather than here? The problem is fairly general. To keep the `solvent_id` specification in the error message and preserve the traceback we could do something like
```python
try:
    warnings = tleap.run()
except RuntimeError as e:
    error = RuntimeError('{}: {}'.format(solvent_id, str(e)))
    error.with_traceback(sys.exc_info()[2])
    raise error
```

Reply:
Will that work as-is?

Reply:
Yeah, I think so!